src/NuGet.Core/NuGet.ProjectModel/DependencyGraphSpec.cs

```diff
@@ -527,5 +531,15 @@ private static string[] GetPackageSpecDependencyIds(PackageSpec spec)
                 .Distinct(PathUtility.GetStringComparerBasedOnOS())
                 .ToArray();
         }
+
+        private static void ConcurrentDictAddCheckDuplicateKey<K, V>(ConcurrentDictionary<K, V> dict, K key, V value)
+        {
+            if (dict.ContainsKey(key))
+            {
+                throw new ArgumentException("An element with the same key already exists in the System.Collections.Generic.Dictionary: " + nameof(dict) + " key:" + key);
```

Comment:
`dict` is the parameter name of the private method, so putting that name in the exception string doesn't help callers of the public API.

Reply:
@zivkan I can just use synchronization lock. @zivkan please check this [PR#3594](https://github.com/NuGet/NuGet.Client/pull/3594).